js/modules/k6/k6.go

```diff
@@ -182,7 +179,9 @@ func (mi *K6) Check(arg0, checks sobek.Value, extras ...sobek.Value) (bool, erro
 		}
 
 		if common.IsAsyncFunction(rt, val) {
-			return false, fmt.Errorf(asyncFunctionNotSupportedMsg, "check")
+			return false, errors.New("built-in check() does not support async functions as arguments, " +
```

Comment:
"The built-in check() does not support async functions as arguments. Use the JavaScript utils library as a replacement. Refer to https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/check/ for more info" 

Small suggestion for the message on these lines.

Reply:
Thanks! I've opened the https://github.com/grafana/k6/pull/4043 for addressing :+1: